ProcessMaker/Jobs/RunScriptTask.php

```diff
@@ -105,13 +105,7 @@ public function action(ProcessRequestToken $token = null, ScriptTaskInterface $e
                 $this->instance = $instance;
             });
         } catch (Throwable $exception) {
-            // Change to error status
-            $token->setStatus(ScriptTaskInterface::TOKEN_STATE_FAILING);
-            $error = $element->getRepository()->createError();
-            $error->setName($exception->getMessage());
-            $token->setProperty('error', $error);
-            Log::error('Script failed: ' . $scriptRef . ' - ' . $exception->getMessage());
-            Log::error($exception->getTraceAsString());
+            $token->logError($exception, $element);
```

Comment:
It seems that when the old lines were removed (108 to 114) when a script task within  a subprocess fails, it is not marked as failing and is left in status ACTIVE. So we have script/service task that never finish or fail and the Retry button  is not  displayed.
The next attached code can be used to reproduce what I describe (Use the start event "Start Event Sub")

[2_scripts.zip](https://github.com/ProcessMaker/processmaker/files/10078479/2_scripts.zip)

Reply:
@danloa Good catch! The only solution I could come up with here was to re-add the lines I took out and I also added in line 115, which seems to work correctly.

Can you confirm? and if it isn't working correctly, what would you recommend? I ask because I'm a bit out of my depth here in working with the more BPMN-intensive portion of the app. :)